src/index.ts

```diff
@@ -1,21 +1,28 @@
-import * as core from '@actions/core';
-import * as exec from '@actions/exec';
-import path from 'path';
+import path from "path";
+import url from "url";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
 
-async function run(): Promise<void> {
-  try {
-    const parentDir = path.resolve(__dirname, '..');
-    
-    // Get action inputs
-    process.env.INPUT_UNSET_PREVIOUS = core.getInput('unset-previous');
-    process.env.INPUT_EXPORT_ENV = core.getInput('export-env');
+const run = async () => {
+	try {
+		const currentFile = url.fileURLToPath(import.meta.url);
+		const currentDir = path.dirname(currentFile);
+		const parentDir = path.resolve(currentDir, "..");
 
-    // Execute bash script
-    await exec.exec(`sh -c "` + parentDir + `/entrypoint.sh"`);
+		// Get action inputs
+		process.env.INPUT_UNSET_PREVIOUS = core.getInput("unset-previous");
+		process.env.INPUT_EXPORT_ENV = core.getInput("export-env");
 
-  } catch (error: any) {
-    core.setFailed(error.message);
-  }
-}
+		// Execute bash script
+		await exec.exec(`sh -c "` + parentDir + `/entrypoint.sh"`);
+	} catch (error) {
+		// https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript
+		let message = "Unknown Error";
+		if (error instanceof Error) {
+			message = error.message;
+		}
+		core.setFailed(message);
+	}
+};
 
-run();
+void run();
```

Comment:
I wonder if there's a real difference between `run()` and `void run()` in this case. 🤔

Reply:
I used void in these instances too, since we're not awaiting anything, there's nothing to `then`, and if it throws we want the error to blow this up